liota/dccs/iotcc.py

```diff
@@ -315,10 +340,26 @@ def set_system_properties(self, reg_entity_obj, system_properties):
             reg_entity_obj.sys_properties = system_properties
             entity = reg_entity_obj.ref_entity
 
-        log.info("System Properties defined for resource {0}".format(entity.name))
+        def on_response(msg):
```

Comment:
Can any of these 'on_response' methods get a msg on the mse_queue that isn't for them? If so, is that message consumed by the particular instances of on_response? If the msg was for another on_response instance is that msg lost? If there are specific instances of the msg_queue for each instance of on_response then why do we even need to test? If the test is required and other msgs can arrive what happens to them?

Reply:
I have updated the pull request for each resource_registration, relationship, properties unregister request to have the separate queue for each call. By doing so we make it thread-safe. I have still kept the checks in on_response to be on safer side.